server/routes/prisons/prisons.test.ts

```diff
@@ -36,335 +36,358 @@ afterEach(() => {
-      })
-  })
-
-  it('should render any error messages set in flash', () => {
-    const error: FieldValidationError = {
-      type: 'field',
-      location: 'body',
-      path: 'prisonId',
-      value: 'X',
-      msg: 'Enter a three letter prison code',
-    }
-    flashData = { errors: [error] }
-
-    return request(app)
-      .get('/prisons')
-      .expect('Content-Type', /html/)
-      .expect(res => {
-        const $ = cheerio.load(res.text)
-        expect($('h1').text().trim()).toBe('Supported prisons')
-        expect($('.govuk-error-summary').text()).toContain(error.msg)
-        expect($('#prisonId-error').text()).toContain(error.msg)
-      })
-  })
-})
-
-describe('POST /prisons', () => {
-  it('should add prison to list of supported prisons and display message', () => {
-    return request(app)
-      .post('/prisons')
-      .send('prisonId=BWI')
-      .expect(302)
-      .expect('location', `/prisons`)
-      .expect(() => {
-        expect(flashProvider).toHaveBeenCalledWith('message', 'BWI')
-        expect(prisonService.createPrison).toHaveBeenCalledTimes(1)
-        expect(prisonService.createPrison).toHaveBeenCalledWith('user1', 'BWI')
-      })
-  })
-
-  it('should set a flash validation error if trying to add a prison which is not in prison register', () => {
-    const error: FieldValidationError = {
-      type: 'field',
-      location: 'body',
-      path: 'prisonId',
-      value: 'XYZ',
-      msg: "Prison 'XYZ' is not in the prison register",
-    }
-
-    prisonService.getPrisonName.mockRejectedValue(new NotFound())
-
-    return request(app)
-      .post('/prisons')
-      .send('prisonId=XYZ')
-      .expect(302)
-      .expect('location', `/prisons`)
-      .expect(() => {
-        expect(flashProvider).toHaveBeenCalledWith('errors', [error])
-        expect(prisonService.createPrison).not.toHaveBeenCalled()
-      })
-  })
-
-  it('should set a flash error if there is an API error response', () => {
-    const error = { msg: '400 Bad Request' }
-    prisonService.createPrison.mockRejectedValue(new BadRequest())
-
-    return request(app)
-      .post('/prisons')
-      .send('prisonId=HEI')
-      .expect(302)
-      .expect('location', `/prisons`)
-      .expect(() => {
-        expect(flashProvider).toHaveBeenCalledWith('errors', [error])
-        expect(prisonService.createPrison).toHaveBeenCalledTimes(1)
-        expect(prisonService.createPrison).toHaveBeenCalledWith('user1', 'HEI')
-      })
-  })
-})
-
-describe('GET /prisons/{:prisonId}/session-templates', () => {
-  const sessionTemplates = [TestData.sessionTemplate()]
-
-  beforeEach(() => {
-    sessionTemplateService.getSessionTemplates.mockResolvedValue(sessionTemplates)
-  })
-
-  describe('Prison status', () => {
-    it('should display prison status information and offer correct action (active prison)', () => {
+describe('Supported prisons page', () => {
```

Comment:
This diff looks like more has changed than is in fact the case! I've basically just shuffled the tests around a bit and added some descriptive `describe()` block wrappers. Hopefully this will make them easier to split up when routes (and consequently tests) refactored in VB-2363.